### ISSUE TITLE
Global config property for default MIME type

### ISSUE THREAD

**Issue by andymc12** [eclipse/microprofile-rest-client#175]
In Issue #111, we declared that MP Rest Client 1.3 interfaces/methods with no `@Produces(...)`/`@Consumes(...)` annotations would default to `application/json`.  It might be good to allow users to override that global default with a MP Config property - for example, `org.eclipse.microprofile.rest.client.defaultProducesMediaType=application/xml` and `org.eclipse.microprofile.rest.client.defaultConsumesMediaType=text/plain`.

Using that example, any interface/method that is not already annotated with `@Produces` would use the specified produces default (`application/xml`), and any interface/method that is not annotated with `@Consumes` would use the the specified consumes default (`text/plain`).  

If these properties are not specified, then unannotated interface/methods would default to `application/json`.

**Comment by kenfinnigan** [comment]
My concern is whether this would make it overly confusing as to what the MIME type will actually be, as there's two possible defaults for a developer to know about instead of one?

**Comment by andymc12** [comment]
Yeah, that is a valid concern.  My main reason for proposing this was to make it easier for users to move from 1.2 (or earlier) to 1.3.  If a user was relying on the previously undefined default MIME type (`application/octet-stream`, `text/plain`, etc.), then they move to 1.3, now they have to update their application code.  If we give them a config switch, then they can migrate the application environment without needing to migrate the code.  

On a similar thought, I was also thinking of opening a separate issue to allow users to override the media types specified in the `@Produces` and `@Consumes` annotations.  This might be a rare case, but my thought was that a client might be accessing a service that changes from one media type to another - if the data model is the same, then it might be nice for the user to add a new config option that changes the media type without needing to update the code.

I definitely don't want to make things too complicated though.  I'm fine with pulling this out of the 1.3 release if you want.

**Comment by kenfinnigan** [comment]
If a developer needs to read docs on what the property is to get back to 1.2 functionality, isn't it easier to document the migration path? Then we don't need additional configuration just to support migration.

As for overriding media types in config, I think we'd want to wait for an actual use case from developers before considering it

**Comment by andymc12** [comment]
I've run into a few customers who have not been able to modify the app code for various reasons, most usually that they got the code from some other organization.  So, while there is a migration path, it might not be as easy as a global switch.  In any case, I'm fine with tabling this issue - if such a customer surfaces, we can address it at that point.

Thanks for the feedback!

**Comment by Emily-Jiang** [comment]
close this since no one provided more feedback.
